feat(prover): support non-mutated verification provider in prover (#6727)
ChainSafe/lodestar: packages/prover/README.md

  --port 8080
```

## How to detect a web3 provider

> There can be different implementations of the web3 providers and each can handle the RPC request differently. We call those different provider types. We had provided builtin support for common providers e.g. web3.js, ethers or any eip1193 compatible providers. We inspect given provider instance on runtime to detect the correct provider type.

Reviewer: ```suggestion
There can be different implementations of the web3 providers and each can handle the RPC request differently. We call those different provider types. We had provided builtin support for common providers e.g. web3.js, ethers or any eip1193 compatible providers. We inspect given provider instance at runtime to detect the correct provider type.
```

> If you project is using some provider type which is not among above list, you have the option to register a custom provider type with the `createVerifiedExecutionProvider` with the option `providerTypes` which will be array of your supported provider types. Your custom provider types will have higher priority on default provider types. Please see [existing provide types implementations](./src/provider_types/) to know how to implement yours own if needed.

Reviewer: ```suggestion
If your project is using some provider type which is not among above list, you have the option to register a custom provider type with the `createVerifiedExecutionProvider` with the option `providerTypes` which will be an array of your supported provider types. Your custom provider types will have higher priority than default provider types. Please see [existing provide types implementations](./src/provider_types/) to know how to implement your own if needed.
```
